plugins/newsdownloader.koplugin/epubdownloadbackend.lua

```diff
@@ -24,6 +24,54 @@ local max_redirects = 5; --prevent infinite redirects
 local TIMEOUT_CODE = "timeout" -- from socket.lua
 local MAXTIME_CODE = "maxtime reached" -- from sink_table_with_maxtime
 
+-- filter HTML using CSS selector
+local function filter(text, element)
+    local htmlparser = require("htmlparser")
+    local root = htmlparser.parse(text)
```

Comment:
Here, I am not sure whether we should use `htmlparser` default looplimit/page depth. With pages with more than 1000 tag depth (which are more popular than expected) the entire thing just does not work and outputs the inserted text. I don't know what value should I set this to then, or to allow user to change it on per feed basis (but then am I not overloading the config?).

Reply:
Is it actually a tag depth of over 1000 or a deadloop because of some invalid HTML that confuses the parser? Please provide a sample of what we're talking about to fully understand the issue.

Assuming it's not a deadloop, the relevant aspect is how much time it takes on an actual device. Is a limit of 10000 still executed in under 100 ms? I'd say that's perfectly fine. Would it take 10 seconds? Maybe consider making it an option. I hope you get the idea, it's meaningless to just talk about the number 1000. ;-)

Reply:
The only case I had this issue was with [BussinessInsider](https://www.businessinsider.com/3-boogaloo-men-terror-charges-george-floyd-protest-riot-conspiracy-2020-6?IR=T). Setting it to 10000 solves the issue, and *on desktop* it takes 0.24 secs, I can check later on my kindle what is the result there.

Reply:
I have set it to 10000 now, and checking on Kindle it seems the slowdown is null so it is a fine resolve.

The complex sites still take a bit longer, but nothing which would disrupt usage.

Reply:
Btw, 10000 was just a somewhat extreme suggestion; what's the actual value on that Business Insider thing? 2500 or so?

Reply:
(Just wondering if it might be safer to go with 5000 to still have a large safety margin, for example.)

Reply:
> Btw, 10000 was just a somewhat extreme suggestion; what's the actual value on that Business Insider thing? 2500 or so?

I checked with some simple js and it seems to be around 1050, in the most extreme case, that being biblegateway it is 2050. I think we can lower it to 5000 and that would leave only edge cases